### DISCUSSION TITLE
2.3.10 install salt key intermittent failure

### DISCUSSION THREAD

**Discussion by darth-drew** [Security-Onion-Solutions/securityonion#2049] [Unsupported Versions]
I have two identical VMs from a hardware resource perspective. One was able to install a distributed manager, airgapped, no problem. The other breaks at 20% mark (Accepting Salt key). Log states "key glob does not match any unaccepted keys".

No other issues detected before this point in the log. Has anyone else run in to this?

**Comment by m0duspwnens** [comment]
@andrewalbertson93 
For the other node that you are trying to install, what is the minion id? This can be found in `/etc/salt/minion` as id:. 

On the manager, do you see any keys listed other than the manager when running `sudo salt-key -l all`?

**Comment by darth-drew** [comment]
So the id is based off of the hostname that was set.. in this case
somanagerdev_manager/somanagertest _manager /somanagerprod _manager. It
does seem like the conflict came from having a manager online and trying to
install another. Once I brought the first one offline the second one was
able to install successfully.

There were no other keys at the time of the error.



On Mon, Nov 23, 2020 at 1:42 PM Josh Patterson <notifications@github.com>
wrote:

> @andrewalbertson93 <https://github.com/andrewalbertson93>
> For the other node that you are trying to install, what is the minion id?
> This can be found in /etc/salt/minion as id:.
>
> On the manager, do you see any keys listed other than the manager when
> running sudo salt-key -l all?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/Security-Onion-Solutions/securityonion/discussions/2049#discussioncomment-136849>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ADC5ZNVOGQIUXTU3OWADWEDSRKUJTANCNFSM4T7YWWVQ>
> .
>


-- 
Andrew R. Albertson
C: (540)908-8551

**Reply by m0duspwnens** [reply]
@andrewalbertson93 
If you are trying to install multiple managers, then it is likely failing for that reason. We plan to add multi manager support in the future.

**Comment by darth-drew** [comment]
To clarify, I am installing 3 separate grids within the same IP space. The
managers are dev/test/prod, rather than hot/warm/cold standby. Does that
make sense? That being said, I was expecting a conflict of this nature.

On Tue, Nov 24, 2020 at 9:55 AM Josh Patterson <notifications@github.com>
wrote:

> @andrewalbertson93 <https://github.com/andrewalbertson93>
> If you are trying to install multiple managers, then it is likely failing
> for that reason. We plan to add multi manager support in the future.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/Security-Onion-Solutions/securityonion/discussions/2049#discussioncomment-137621>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ADC5ZNQWW7YEL3YDFHJWUB3SRPCOLANCNFSM4T7YWWVQ>
> .
>


-- 
Andrew R. Albertson
C: (540)908-8551

**Comment by darth-drew** [comment]
was not*


On Tue, Nov 24, 2020 at 10:14 AM Andrew Albertson <
andrew.albertson93@gmail.com> wrote:

> To clarify, I am installing 3 separate grids within the same IP space. The
> managers are dev/test/prod, rather than hot/warm/cold standby. Does that
> make sense? That being said, I was expecting a conflict of this nature.
>
> On Tue, Nov 24, 2020 at 9:55 AM Josh Patterson <notifications@github.com>
> wrote:
>
>> @andrewalbertson93 <https://github.com/andrewalbertson93>
>> If you are trying to install multiple managers, then it is likely failing
>> for that reason. We plan to add multi manager support in the future.
>>
>> —
>> You are receiving this because you were mentioned.
>> Reply to this email directly, view it on GitHub
>> <https://github.com/Security-Onion-Solutions/securityonion/discussions/2049#discussioncomment-137621>,
>> or unsubscribe
>> <https://github.com/notifications/unsubscribe-auth/ADC5ZNQWW7YEL3YDFHJWUB3SRPCOLANCNFSM4T7YWWVQ>
>> .
>>
>
>
> --
> Andrew R. Albertson
> C: (540)908-8551
>


-- 
Andrew R. Albertson
C: (540)908-8551

**Comment by m0duspwnens** [comment]
Ok, I want to make sure I understand the scenario correctly. You are trying to install 3 managers, and the first manager has installed properly. When you try to install the second manager, it fails at 20% if the other manager is also online? If the first manager is offline, it installs properly. Is that correct?

For the manager that is failing at 20% does it say `Accepting Salt key on manager` or does it say `Accepting Salt key`? 

When you run `salt-key -l all` on the manager that has failed, does it list any keys?

On the manager that failed, if you run `salt-call state.show_top`, what output do you get?

In our lab we run multiple managers within the same subnet and haven't seen this issue.

**Comment by darth-drew** [comment]
The scenario you described is accurate. The second breaks at 20% mark (Accepting Salt key). Log states "key glob does not match any unaccepted keys". There were no other keys at the time of the error.

Unfortunately, I won't be able to provide the show_top output at this time... the reinstall was already completed. The workaround is sufficient for our purposes. If time allows, I will try and replicate and provide further details to you guys. Feel free to take this out of your queue!

**Reply by defensivedepth** [reply]
Were the VMs cloned? Wondering if the mac addresses need to be regenerated.
